python/ray/air/util/torch_dist.py

```diff
@@ -0,0 +1,158 @@
+Eventually, these use cases should be consolidated.
+"""
+
+from abc import ABC
+from datetime import timedelta
+import os
+import torch.distributed as dist
+from typing import Callable, List, T
+
+import ray
+from ray.actor import ActorHandle
+from ray.air._internal.util import skip_exceptions, exception_cause
+from ray.train._internal.utils import get_address_and_port
+from ray.train.constants import DEFAULT_NCCL_SOCKET_IFNAME
+
+
+class TorchDistributedWorker(ABC):
+    """Defines the interfaces required by the `init_torch_dist_process_group()`.
+
+    This is modeled after RayTrainerWorker, which allows arbitrary functions
+    to be executed on a remote DDP worker.
+    """
+
+    def execute(self, func: Callable[..., T], *args, **kwargs) -> T:
+        """Executes the input function and returns the output.
+
+        Args:
+            func: The function to execute.
+            args, kwargs: The arguments to pass into func.
+        """
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            skipped = skip_exceptions(e)
+            raise skipped from exception_cause(skipped)
+
+
+def _init_torch_distributed(
+    init_method: str,
+    backend: str,
+    rank: int,
+    world_size: int,
+    local_rank: int,
+    local_world_size: int,
+    master_addr: str,
+    master_port: str,
+    gpu_ids: List[int],
+):
+    """Initialize torch distributed backend"""
+    if init_method == "env":
+        os.environ["MASTER_ADDR"] = str(master_addr)
+        os.environ["MASTER_PORT"] = str(master_port)
+        url = "env://"
+    elif init_method == "tcp":
+        url = f"tcp://{master_addr}:{master_port}"
+    else:
+        raise ValueError(
+            f"The provided init_method ("
+            f"{init_method}) is not supported. Must "
+            f"be either 'env' or 'tcp'."
+        )
+
+    if backend == "nccl":
+        # Same as in Ray Train
+        os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "1"
+        # All workers on a same node should share the same set of
+        # visible GPUs. Otherwise they can't talk among themselves.
+        os.environ["CUDA_VISIBLE_DEVICES"] = ",".join(str(gid) for gid in gpu_ids)
+        if "NCCL_SOCKET_IFNAME" not in os.environ:
+            os.environ["NCCL_SOCKET_IFNAME"] = DEFAULT_NCCL_SOCKET_IFNAME
+
+    dist.init_process_group(
+        backend=backend,
+        init_method=url,
+        rank=rank,
+        world_size=world_size,
+        timeout=timedelta(seconds=1800),
+    )
+
+    os.environ["RANK"] = str(rank)
+    os.environ["LOCAL_RANK"] = str(local_rank)
+    os.environ["WORLD_SIZE"] = str(world_size)
+    os.environ["LOCAL_WORLD_SIZE"] = str(local_world_size)
+
+
+def _get_node_and_gpu_ids():
+    """Returns the node_id and gpu_ids for this worker."""
+    node_id = ray.get_runtime_context().get_node_id()
+    gpu_ids = ray.get_gpu_ids()
+    return node_id, gpu_ids
+
+
+def init_torch_dist_process_group(
```

Comment:
is this intended to be public API? If not, then let's move it into `_internal` package?

Reply:
this will be included in our public facing examples.
if we move those predictors into ray/air/, we can mark this private, or get rid of this altogether and refactor Train to share this logic.